event-handler/teleport_events_watcher_test.go

```diff
@@ -108,3 +109,97 @@ func TestNext(t *testing.T) {
 		require.Fail(t, "No events were sent")
 	}
 }
+
+func TestValidateConfig(t *testing.T) {
+	tempDir := t.TempDir()
+	tempFile, err := os.CreateTemp(tempDir, "temp_file")
+	require.NoError(t, err)
+
+	for _, tc := range []struct {
+		name         string
+		cfg          StartCmdConfig
+		errorMessage string
+	}{
+		{
+			name: "All set and exist",
+			cfg: StartCmdConfig{
+				FluentdConfig{
+					FluentdCert: tempFile.Name(),
+					FluentdKey:  tempFile.Name(),
+					FluentdCA:   tempFile.Name(),
+				},
+				TeleportConfig{
+					TeleportIdentityFile: tempFile.Name(),
+					TeleportCA:           tempFile.Name(),
+					TeleportCert:         tempFile.Name(),
+					TeleportKey:          tempFile.Name(),
+				},
+				IngestConfig{},
+				LockConfig{},
+			},
+		}, {
+			name: "All set and some do not exist",
+			cfg: StartCmdConfig{
+				FluentdConfig{
+					FluentdCert: "tempFile.Name()",
+					FluentdKey:  tempFile.Name(),
+					FluentdCA:   tempFile.Name(),
+				},
+				TeleportConfig{
+					TeleportIdentityFile: tempFile.Name(),
+					TeleportCA:           tempFile.Name(),
+					TeleportCert:         tempFile.Name(),
+					TeleportKey:          tempFile.Name(),
+				},
+				IngestConfig{},
+				LockConfig{},
+			},
+			errorMessage: "required file \"tempFile.Name()\" does not exist",
+		}, {
+			name: "Some not set and all exist",
+			cfg: StartCmdConfig{
+				FluentdConfig{
+					FluentdCert: tempFile.Name(),
+					FluentdKey:  tempFile.Name(),
+					FluentdCA:   tempFile.Name(),
+				},
+				TeleportConfig{
+					TeleportIdentityFile: tempFile.Name(),
+					TeleportCA:           tempFile.Name(),
+					// TeleportCert:         tempFile.Name(),
+					// TeleportKey:          tempFile.Name(),
+				},
+				IngestConfig{},
+				LockConfig{},
+			},
+			errorMessage: "required config field \"TeleportCert\" not set, required config field \"TeleportKey\" not set",
+		}, {
+			name: "Some not set and some do not exist",
+			cfg: StartCmdConfig{
+				FluentdConfig{
+					FluentdCert: tempFile.Name(),
+					FluentdKey:  "tempFile.Name()",
+					FluentdCA:   tempFile.Name(),
+				},
+				TeleportConfig{
+					TeleportIdentityFile: tempFile.Name(),
+					TeleportCA:           tempFile.Name(),
+					TeleportCert:         tempFile.Name(),
+					// TeleportKey:          tempFile.Name(),
+				},
+				IngestConfig{},
+				LockConfig{},
+			},
+			errorMessage: "required file \"tempFile.Name()\" does not exist, required config field \"TeleportKey\" not set",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			err := validateConfig(tc.cfg)
+			if tc.errorMessage != "" {
+				require.EqualError(t, err, tc.errorMessage)
```

Comment:
I wouldn't assert on a specific error message - this is easy to break if we change the wording of an error.

In general, I would expect that invalid configuration would result in a `trace.BadParameter` error - so how about we just assert that `trace.IsBadParameter(err)` is true?